src/context/provider.tsx

```diff
@@ -1,19 +1,36 @@
-import React, { ReactElement, ReactNode } from 'react';
-import { useQuery, UseQueryOptions } from 'react-query';
+import React, { ReactElement, ReactNode, useEffect, useState } from 'react';
 
-import { SettingsContext, SettingsProviderValue } from '@/context/context';
+import { SettingsContext, SettingsProviderValue } from './context';
 
 export type SettingsValue = Pick<SettingsProviderValue, 'settings' | 'flags'>;
 
 interface Props {
   children: ReactNode;
   getSettings: () => Promise<SettingsValue>;
-  options?: UseQueryOptions<SettingsValue>;
+  updateIntervalMs?: number;
 }
 
-export function SettingsProvider({ children, getSettings, options = {} }: Props): ReactElement<Props> {
-  const { data, isLoading } = useQuery('react-settings-provider', getSettings, options);
-  const settings = data?.settings ?? {};
-  const flags = data?.flags ?? {};
-  return <SettingsContext.Provider value={{ settings, flags, isLoading }}>{children}</SettingsContext.Provider>;
+export function SettingsProvider({ children, getSettings, updateIntervalMs }: Props): ReactElement<Props> {
+  const [isLoading, setLoading] = useState(false);
+  const [value, setValue] = useState<SettingsValue>({ flags: {}, settings: {} });
+
+  useEffect(() => {
+    function load() {
+      if (isLoading) return;
+      setLoading(true);
+      getSettings()
+        .then(setValue)
```

Comment:
**question**: should we clean `value` if `getSettings` rejects? What would be less harmful in this case: keep as is or use an outdated version of settings (one that might have disabled a flag or remove a setting, for example)?

Reply:
Hm, that's a valid point, IMO it would be better to use an old value if available as it would be less disruptive, but maybe we could add an error handler that the user could pass to the provider, what do you think?

Reply:
That sounds good to me